### DISCUSSION TITLE
Missing version constraint info

### DISCUSSION THREAD

**Discussion by elsazac** [eclipse-platform/eclipse.platform.ui#1588] [General]
I see this warning for many bundles : platform ui, pde etc. Is there a best practice around version constraints and is there a known solution to solve it? Is there a tradeoff for leaving bundle versions unspecified ?

**Comment by merks** [comment]
Of course semantically it should be the lowest version for which the bundle functions correctly for that dependency.  So often it's set to the current version, once, in the beginning because PDE helpfully does that automatically in the editor.  But a problemfor package requirements is that those mostly aren't properly versioned so one cannot specify a version.  Another problem is that the lower bound is generally not updated when it should be updated because there is no tooling that tells you that you should update it.  So even when present, it's not necessarily correct and in fact most likely 95% of them are wrong.   As such, like the 6000 other warnings in the SDK workspace, folks tend to ignore  the issue.

**Reply by merks** [reply]
You'll probably want to read this:

https://github.com/eclipse-platform/eclipse.platform/blob/master/docs/VersionNumbering.md

**Comment by laeubi** [comment]
I'm currently working on a mojo to automate these tasks here:
- https://github.com/eclipse-tycho/tycho/pull/3427

I think that's the only way forward (beside using automatic manifest in the first place), I though about changing Tycho itself to support such cases where one wants to be compatible for a range of dependencies but thats nothing happen in the near future (unless there is some higher demands):
- https://github.com/eclipse-tycho/tycho/issues/2259
